sympy/physics/mechanics/functions.py

```diff
@@ -532,7 +539,7 @@ def _recurser(expr, sub_dict):
 
 def _fraction_decomp(expr):
     """Return num, den such that expr = num/den"""
-    if not isinstance(expr, Mul):
+    if not isinstance(expr, MulClass):
```

Comment:
It could be part of either library.

It technically makes a huge difference to me. But it seems we both agree that we need a 100% (or close to 100%) compatibility layer, somewhere. Whether built on top of symengine.py, or whether symengine.py itself (i.e. the Cython wrapper) would be 100% compatible, those are various technical solutions to the same, ultimate problem.

We should discuss what the compatibility layer should have. E.g. what I have in mind is that if you do things like

a = sin(x)
b = a.diff(x)

Then "b" will have exactly the same methods, whether you call it with sympy or symengine backend.

I am worried about speed implications caused by this compatibility layer. I can see pros and cons of each technical solution that I have in mind.

Right now I lean towards just making our best effort to clean this PR, to make it reasonable clean to merge, and then investigate long term solutions.